### PR TITLE
Enhance interface class and LinuxInterface

### PR DESCRIPTION
**Pull request by pichuang**
1. Detect interface connected state of a network cable/connector or not
2. Get MTU on the interface
3. Add python docstring into interface class

Signed-off-by: Phil Huang <phil_huang@edge-core.com>

### CODE REVIEW

File: testinfra/modules/interface.py [hunk 1]
```diff
@@ -25,10 +25,20 @@ def __init__(self, name):
 
     @property
     def exists(self):
+        """Return logical interface is exist or not
+
+        >>> host.interface("eth0").exists
+        1
```
Comment (philpep) [inline testinfra/modules/interface.py]: This should be True or False

File: testinfra/modules/interface.py [hunk 2]
```diff
@@ -25,10 +25,20 @@ def __init__(self, name):
 
     @property
     def exists(self):
+        """Return logical interface is exist or not
```
Comment (philpep) [inline testinfra/modules/interface.py]: maybe "Return if logical interface exists or not" ?

File: testinfra/modules/interface.py [hunk 3]
```diff
@@ -25,10 +25,20 @@ def __init__(self, name):
...
+
+        >>> host.interface("eth0").exists
+        1
+        """
         raise NotImplementedError
 
     @property
     def speed(self):
+        """Return port speed on the interface
```
Comment (philpep) [inline testinfra/modules/interface.py]: Maybe adding "in Mb/s"

File: testinfra/modules/interface.py [hunk 4]
```diff
@@ -40,6 +50,50 @@ def addresses(self):
         """
         raise NotImplementedError
 
+    @property
+    def is_up(self):
+        """Return detect interface connected state of a network cable/connector or not
+
+        >>> host.interface("eth0").is_up
+        1
```
Comment (philpep) [inline testinfra/modules/interface.py]: True/False

File: testinfra/modules/interface.py [hunk 5]
```diff
@@ -40,6 +50,50 @@ def addresses(self):
...
+        """
+        raise NotImplementedError
+
+    @property
+    def mtu(self):
+        """Return MTU on the interface
```
Comment (philpep) [inline testinfra/modules/interface.py]: add "in bytes"

File: testinfra/modules/interface.py [hunk 6]
```diff
@@ -40,6 +50,50 @@ def addresses(self):
...
+        1500
+        """
+        raise NotImplementedError
+
+    @property
+    def is_bond(self):
+        """Return the interface type is bond or not
+
+        >>> host.interface("eth0").is_bond
+        0
```
Comment (philpep) [inline testinfra/modules/interface.py]: True/False

File: testinfra/modules/interface.py [hunk 7]
```diff
@@ -40,6 +50,50 @@ def addresses(self):
...
+        """
+        raise NotImplementedError
+
+    @property
+    def is_bridge(self):
+        """Return the interface type is bridge or not
+
+        >>> host.interface("eth0").is_bridge
+        0
```
Comment (philpep) [inline testinfra/modules/interface.py]: True/False

File: testinfra/modules/interface.py [hunk 8]
```diff
@@ -40,6 +50,50 @@ def addresses(self):
...
+        """
+        raise NotImplementedError
+
+    def is_vlan(self):
+        """Return the interface type is vlan or not
+
+        >>> host.interface("eth0").is_vlan
+        0
```
Comment (philpep) [inline testinfra/modules/interface.py]: True/False

**Comment by pichuang** [comment]
Hi @philpep  Please review it.

**Comment by pichuang** [comment]
Hi @philpep  Could you please review this change? Thanks~

File: testinfra/modules/interface.py [hunk 9]
```diff
@@ -74,6 +128,37 @@ def addresses(self):
...
+            return False
+
+    @property
+    def mtu(self):
+        return int(self.check_output(
+            "cat /sys/class/net/%s/mtu", self.name))
+
+    @property
+    def is_bond(self):
+        return self.run_test("ip link show %s type bond", self.name).rc == 0
```
Comment (philpep) [inline testinfra/modules/interface.py]: (Sorry for delay) I think the command is wrong:

```
$ ip link show eth0 type bond
Error: either "dev" is duplicate, or "type" is a garbage.
$ echo $?
255
```

Ideally this deserve tests, for example in test/test_modules.py:

```python
def test_interface(host):
    iface = host.interface("eth0")
    assert iface.is_up
    assert iface.mtu > 0
    assert iface.speed > 0
    assert not iface.is_bond
    assert not iface.is_bridge
    assert not iface.is_vlan
```

**Comment by pichuang** [comment]
HI @philpep Come back again. Could you please review?

File: testinfra/modules/interface.py [hunk 10]
```diff
@@ -74,6 +119,37 @@ def addresses(self):
...
+    @property
+    def mtu(self):
+        return int(self.check_output(
+            "cat /sys/class/net/%s/mtu", self.name))
+
+    @property
+    def is_bridge(self):
+        return self.run_expect([0, 1, 255],
+                               "ip link show %s type bridge",
+                               self.name).rc == 0
```
Comment (tyll) [inline testinfra/modules/interface.py]: It seems that you could just do `os.path.isdir("/sys/class/net/%s/bridge" % self.name` here.

**Pull request closed without merging**